### PR TITLE
[FEAT] Enable opt-in option for account discount

### LINKED ISSUE #4: [FEAT] Enable opt-in option for account discount
### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Hi

Azure Cosmosdb support one free account per subscription;

This is an [opt-in feature](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/cosmosdb_account#enable_free_tier) and in order to enable it, we need to set `enable_free_tier = true` inside the module. 

There is no complicated condition about this capability. So, I think we can just simply enable it using a new variable `var.enable_free_tier` with default value set to `false`.

I am happy to add this to the module.

### New or Affected Resource(s)/Data Source(s)

azurerm_cosmosdb_account

### Potential Terraform Configuration

_No response_

### References

https://learn.microsoft.com/en-us/azure/cosmos-db/free-tier#free-tier-with-azure-discount
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/cosmosdb_account#enable_free_tier

### PR DESCRIPTION
**Pull request by kevin-fitlg**
Fixes #4 .

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Changes proposed in this pull request

- add variable to support free account option (opt-in)

@claranet/fr-azure-reviewers

### CODE REVIEW

**Comment by kevin-fitlg** [comment]
Hey team, first time I am opening a PR on a public repo :) 
So, please let me know if there is any issue.
Thanks

File: r-cosmosdb.tf [hunk 1]
```diff
@@ -7,6 +7,7 @@ resource "azurerm_cosmosdb_account" "db" {
   offer_type           = var.offer_type
   kind                 = var.kind
   mongo_server_version = var.kind == "MongoDB" ? var.mongo_server_version : null
+  enable_free_tier     = var.enable_free_tier
```
Comment (jmapro) [inline r-cosmosdb.tf]: To keep consistency with other modules, for boolean variables we put "enabled" at the end of the variable name.

```suggestion
  enable_free_tier     = var.free_tier_enabled
```

File: variables.tf [hunk 2]
```diff
@@ -165,3 +165,9 @@ variable "identity_type" {
   type        = string
   default     = "SystemAssigned"
 }
+
+variable "enable_free_tier" {
```
Comment (jmapro) [inline variables.tf]:
```suggestion
variable "free_tier_enabled" {
```

**Review by jmapro — CHANGES_REQUESTED** [review]

**Comment by kevin-fitlg** [comment]
Thanks @jmapro 
Suggestions applied;

**Review by jmapro — APPROVED** [review]
Thanks for your contribution !

LGTM

**Review by BzSpi — APPROVED** [review]

**Comment by BzSpi** [comment]
Thank you for your contribution

**Comment by kevin-fitlg** [comment]
Hey @BzSpi / @jmapro 
When do you generally release these changes, i.e. add a new tag to the repo? (I need this attribute in one my deployments)

**Comment by BzSpi** [comment]
Within this week for sure, it can be today.

**Comment by rossifumax** [comment]
Hello @kevin-fitlg !

Thanks for your contribution :+1: 

Tag [v6.4.0](https://github.com/claranet/terraform-azurerm-cosmos-db/tree/v6.4.0) is available :tada:

**Pull request merged**